cluster-autoscaler/main.go

```diff
@@ -465,12 +465,7 @@ func registerSignalHandlers(autoscaler core.Autoscaler) {
 	}()
 }
 
-func buildAutoscaler(debuggingSnapshotter debuggingsnapshot.DebuggingSnapshotter) (core.Autoscaler, error) {
-	// Create basic config from flags.
-	autoscalingOptions := createAutoscalingOptions()
-
-	autoscalingOptions.KubeClientOpts.KubeClientBurst = int(*kubeClientBurst)
-	autoscalingOptions.KubeClientOpts.KubeClientQPS = float32(*kubeClientQPS)
+func buildAutoscaler(debuggingSnapshotter debuggingsnapshot.DebuggingSnapshotter, autoscalingOptions config.AutoscalingOptions, restConfig *rest.Config, injector pods.PodListProcessor) (core.Autoscaler, error) {
```

Comment:
I'm not sure that buildAutoscaler should depend on provReq injector. ProvisioningRequest is a feature rather that global cluster setting. 

We can do is introduce LoopTriger field in AutoscalerOptions and pass injector there.

Reply:
Updated such that buildAutoscaler returns trigger